src/TritonRoute/src/TritonRoute.tcl

```diff
@@ -34,17 +34,120 @@
 #############################################################################
 
 sta::define_cmd_args "detailed_route" {
-    -param filename
+    -guide filename
+    [-outputguide filename]
+    [-outputmaze filename]
+    [-outputDRC filename]
+    [-outputCMap filename]
+    [-dbProcessNode name]
+    [-drouteEndIterNum iter]
+    [-drouteViaInPinBottomLayerNum num]
+    [-drouteViaInPinTopLayerNum num]
+    [-OR_SEED seed]
+    [-OR_K k]
+    [-bottomRoutingLayer layer]
+    [-topRoutingLayer layer]
+    [-initRouteShapeCost cost]
+    [-verbose level]
+    [-param filename]
 }
 
 proc detailed_route { args } {
-  sta::parse_key_args "detailed_route" args keys {-param}
+  sta::parse_key_args "detailed_route" args \
+    keys {-param -guide -outputguide -outputmaze -outputDRC -outputCMap \
+      -dbProcessNode -drouteEndIterNum -drouteViaInPinBottomLayerNum \
+      -drouteViaInPinTopLayerNum -OR_SEED -OR_K -bottomRoutingLayer \
+      -topRoutingLayer -initRouteShapeCost -verbose}
   sta::check_argc_eq0 "detailed_route" $args
 
-  if { ![info exists keys(-param)] } {
+  if { ![info exists keys(-guide)] && ![info exists keys(-param)] } {
```

Comment:
-guide isn't mandatory.  When I integrated TR-WXL we got a global router that turns on when -guide isn't given.

Reply:
Oh nice, I didn't know that.

Reply:
@maliberty it looks like if I don't provide the guide file, GRT does not get run and I get a bunch of
```
Error: non-root node does not have parent
```

Reply:
If you look at
int TritonRoute::main()
{
  init();
  if (GUIDE_FILE == string("")) {
    gr();

So it should run gr if not found.  I have tested it in the past and it worked though we don't have a regular regression for it.

Reply:
@maliberty When running through GDB I find that it calls `triton_route::TritonRoute::gr`, are you trying to call `grt::run`?

Reply:
I find that gr is getting run, but that is actually what produces all the errors.